src/kemal/static_file_handler.cr

```diff
@@ -76,6 +79,65 @@ module Kemal
       end
     end
 
+    def multipart(file, env)
+      #See http://httpwg.org/specs/rfc7233.html
+      fileb = file.size
+
+      range = env.request.headers["Range"]
+      match = range.match(/bytes=(\d{1,})-(\d{0,})/)
+
+      startb = 0
+      endb = 0
+
+      if match
+        if match.size >= 2
+          startb = match[1].to_i { 0 }
+        end
+
+        if match.size >= 3
+          endb = match[2].to_i { 0 }
+        end
+      end
+
+      if endb == 0
+        endb = fileb
+      end
+
+      if startb < endb && endb <= fileb
+        env.response.status_code = 206
+        env.response.content_length = endb - startb
+        env.response.headers.add("Content-Range", "bytes #{startb}-#{endb - 1}/#{fileb}") # MUST
```

Comment:
is there any particular reason you're calling `#add` here, instead of `#[]=` (like for `X-Streaming` below)? i'm just curious :)